sklearn/neighbors/_kd_tree.pyx.tp

```diff
@@ -240,8 +317,19 @@ cdef inline float64_t max_rdist_dual(BinaryTree tree1, intp_t i_node1,
     return rdist
 
 
-cdef inline float64_t max_dist_dual(BinaryTree tree1, intp_t i_node1,
-                                    BinaryTree tree2, intp_t i_node2) except -1:
+cdef inline float64_t max_dist_dual{{name_suffix}}(
+    BinaryTree{{name_suffix}} tree1,
+    intp_t i_node1,
+    BinaryTree{{name_suffix}} tree2,
+    intp_t i_node2,
+) except -1:
     """Compute the maximum distance between two nodes"""
-    return tree1.dist_metric._rdist_to_dist(max_rdist_dual(tree1, i_node1,
-                                                           tree2, i_node2))
+    return tree1.dist_metric._rdist_to_dist(
+        max_rdist_dual{{name_suffix}}(tree1, i_node1, tree2, i_node2)
+    )
+
+{{endfor}}
+
+
+class KDTree(KDTree64):
+    pass
```

Comment:
Does this need to be a subclass?

```suggestion
KDTree = KDTree64
```

Reply:
I think so because:

```python
In [1]: class A:
   ...:     pass
   ...: 

In [2]: B = A

In [3]: B()
Out[3]: <__main__.A at 0x7fb36fdd0100>
```

Reply:
I see what you mean. In that case, I am okay with sublcassing.